src/number_theory/bertrand.lean

```diff
@@ -0,0 +1,548 @@
+groups, then upper bounds the contribution of each group. This upper bounds the central binomial
+coefficient, and if the postulate does not hold, this upper bound conflicts with a simple lower
+bound for large enough `n`. This proves the result holds for large enough `n`, and for smaller `n`
+an explicit list of primes is provided which covers the remaining cases.
+
+As in the [Metamath implementation](carneiro2015arithmetic), we rely on some optimizations from
+[Shigenori Tochiori](tochiori_bertrand). In particular we use the fact that `(log x) / x` is
+decreasing for `e ≤ x`.
+
+## Proof Sketch
+
+Here is a description of how the proof works:
+
+Then:
+4^n ≤ 2nCn * (2 * n + 1) (by choose_halfway_is_big)
+= prod (primes ≤ 2n) p^(α n p) * (2n+1)
+= prod (primes ≤ n) p^(α n p) * prod (primes n < p <= 2n) p^α * (2n+1)
+= prod (primes ≤ 2n/3) p^α * prod (primes 2n/3 to n) p^α * prod (primes n < p ≤ 2n) p^α * (2n+1)
+= prod (primes ≤ 2n/3) p^α * prod (primes 2n/3 to n) 1 * prod (primes n < p ≤ 2n) p^α * (2n+1)
+  (by claim 3)
+= prod (primes ≤ 2n/3) p^α * prod (primes n < p ≤ 2n) p^α * (2n+1)
+= prod (primes ≤ sqrt(2n)) p^α * prod(primes sqrt(2n)..2n/3) p^α
+  * prod (primes n < p ≤ 2n) p^α * (2n+1)
+≤ prod (primes ≤ sqrt(2n)) p^α * prod(primes sqrt(2n)..2n/3) p
+  * prod (primes n < p ≤ 2n) p^α * (2n+1)
+  (by claim 2)
+≤ prod (primes ≤ sqrt(2n)) p^α * 4 ^ (2n / 3) * prod (primes n < p ≤ 2n) p^α * (2n+1)
+  (by a general bound on the primorial)
+≤ prod (primes ≤ sqrt(2n)) (2n) * 4 ^ (2n / 3) * prod (primes n < p ≤ 2n) p^α * (2n+1)
+  (by claim 1)
+= (2n)^π (sqrt 2n) * 4 ^ (2n/3) * prod (primes n < p ≤ 2n) p^α * (2n+1)
+≤ (2n)^(sqrt 2n) * 4 ^ (2n/3) * prod (primes n < p ≤ 2n) p^α * (2n+1)
+  (by "prime count of x is less than x")
+
+For sufficiently large n, that last product term is > 1.
+Indeed, suppose for contradiction it's equal to 1.
+Then 4^n ≤ (2n)^(sqrt 2n) * 4^(2n/3) * (2n+1)
+so 4^(n/3) ≤ (2n)^(sqrt 2n) (2n+1)
+and this is Clearly False for sufficiently large n.
+
+## References
+
+* [M. Aigner and G. M. Ziegler _Proofs from THE BOOK_][aigner1999proofs]
+* [S. Tochiori, _Considering the Proof of “There is a Prime between n and 2n”_][tochiori_bertrand]
+* [M. Carneiro, _Arithmetic in Metamath, Case Study: Bertrand's Postulate_][carneiro2015arithmetic]
+-/
+
+open_locale big_operators
+
+section real_inequalities
+open real
+
+lemma log_four_pos : 0 < log 4 := log_pos (by linarith)
+
+lemma log_four_nonzero : log 4 ≠ 0 := by linarith [log_four_pos]
+
+lemma log_1024_div_log_4 : log 1024 / log 4 = 5 :=
+begin
+  have h : (1024 : ℝ) = 4 ^ (5 : ℝ) := by norm_num,
+  rw [div_eq_iff log_four_nonzero, h, log_rpow],
+  linarith,
+end
+
+lemma four_eq_two_rpow_two : (4 : ℝ) = 2 ^ (2 : ℝ) := by norm_num
+
+lemma log_four : log 4 = 2 * log 2 :=
+calc log 4 = log (2 ^ (2 : ℝ)) : by norm_num
+  ... = 2 * log 2 : by rw log_rpow two_pos
+
+lemma sqrt_361 : sqrt 361 = 19 :=
```

Comment:
`norm_num` should be able to do this now? or is this `real.sqrt` and therefore it doesn't work?

Reply:
It's real, but I inlined it anyway